src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs

```diff
@@ -103,27 +116,42 @@ private static unsafe SafeMsQuicConfigurationHandle Create(QuicOptions options,
                 // MsQuic has a private copy of OpenSSL so the SSL_CTX will be incompatible.
 
                 CredentialConfig config = default;
-
                 config.Flags = flags; // TODO: consider using LOAD_ASYNCHRONOUS with a callback.
 
                 if (certificate != null)
                 {
-#if true
-                    // If using stub TLS.
-                    config.Type = QUIC_CREDENTIAL_TYPE.STUB_NULL;
-#else
-					// TODO: doesn't work on non-Windows
-                    config.Type = QUIC_CREDENTIAL_TYPE.CONTEXT;
-                    config.Certificate = certificate.Handle;
-#endif
+                    // TODO: doesn't work on non-Windows
+                    if (OperatingSystem.IsWindows())
+                    {
+                        config.Type = QUIC_CREDENTIAL_TYPE.CONTEXT;
+                        config.Certificate = certificate.Handle;
+                        status = MsQuicApi.Api.ConfigurationLoadCredentialDelegate(configurationHandle, ref config);
+                    }
+                    else
+                    {
+                        CredentialConfigCertificatePkcs12 pkcs12Config;
+                        byte[] asn1 = certificate.Export(X509ContentType.Pkcs12);
+                        unsafe
+                        {
+                            fixed (void* ptr = asn1)
```

Comment:
I think this is the only occurrence of `fixed` in here. I'm not an interop expert, but should we rather use `GCHandle` as we do in other places? Is there any difference in those two? If not should we rather stick to the same approach throughout?
cc: @scalablecory

Reply:
You need to allocate GCHandle to preserve object for longer time. This basically says "don't move asn1 while I'm in this block" and it is much simpler. (and does not work for the other cases.